src/check-custom-fields.ts

```diff
@@ -0,0 +1,64 @@
+import * as core from '@actions/core'
+
+interface ObjectType {
+  [key: string]: string
+}
+
+interface OptionType {
+  id: string
+  name: string
+}
+
+interface ArrayType {
+  options: OptionType[]
+  name: string
+  dataType: string
+  id: string
+}
+
+export async function checkDictionaryInArray(obj: {[key: string]: string}, arr: []) {
+  const object: ObjectType = obj
+  const array: ArrayType[] = arr
+
+  const result = []
+
+  //remove empty objects from array of objects
+  const filteredArray = array.filter(objs => !(objs && Object.keys(objs).length === 0 && objs.constructor === Object))
```

Comment:
Great. A couple of comments:

- [ ] Let's rename `filteredArray` to `clearedProjectFields`
- [ ] Let's move the filter function to a named function (e.g. `isNotEmpty`):
    ```ts
    const isNotEmpty = obj => !(obj && Object.keys(obj).length === 0 && obj.constructor === Object)
    ```